training/src/main/scala/com/airbnb/aerosolve/training/AdditiveModelTrainer.scala

```diff
@@ -255,7 +259,7 @@ object AdditiveModelTrainer {
     val expCorr = scala.math.exp(corr)
     val loss = scala.math.log(1.0 + 1.0 / expCorr)
     val grad = -label / (1.0 + expCorr)
-    val gradWithLearningRate = grad.toFloat * params.learningRate.toFloat
+    val gradWithLearningRate = grad.toFloat * params.learningRate.toFloat * classWeights(label.toInt)
```

Comment:
I think this line is under `updateLogistic` :), and you can find another line below for `updateHinge`